src/SampleViewExtension/SampleViewExtension.cs

```diff
@@ -9,69 +9,72 @@ namespace SampleViewExtension
+    /// IViewExtension or extends ViewExtensionBase, and a ViewExtensionDefinition
+    /// XML file used to instruct Dynamo where to find the class containing the
+    /// View Extension implementation. The ViewExtensionDefinition XML file must
     /// be located in your [dynamo]\viewExtensions folder.
     /// 
-    /// This sample demonstrates an IViewExtension implementation which 
-    /// shows a modeless window when its MenuItem is clicked. 
+    /// This sample demonstrates a View Extension implementation which inherits from
+    /// ViewExtensionBase. It adds a user interface to Dynamo's extension sidebar
+    /// when its MenuItem is clicked. 
     /// The Window created tracks the number of nodes in the current workspace, 
     /// by handling the workspace's NodeAdded and NodeRemoved events.
     /// </summary>
-    public class SampleViewExtension : IViewExtension
+    public class SampleViewExtension : ViewExtensionBase
     {
         private MenuItem sampleMenuItem;
 
-        public void Dispose()
+        public override void Dispose()
         {
         }
 
-        public void Startup(ViewStartupParams p)
+        public override void Startup(ViewStartupParams p)
         {
         }
 
-        public void Loaded(ViewLoadedParams p)
+        public override void Loaded(ViewLoadedParams p)
         {
             // Save a reference to your loaded parameters.
             // You'll need these later when you want to use
             // the supplied workspaces
 
-            sampleMenuItem = new MenuItem {Header = "Show View Extension Sample Window"};
-            sampleMenuItem.Click += (sender, args) =>
+            var viewModel = new SampleWindowViewModel(p);
+            var window = new SampleWindow
             {
-                var viewModel = new SampleWindowViewModel(p);
-                var window = new SampleWindow
-                {
-                    // Set the data context for the main grid in the window.
-                    MainGrid = { DataContext = viewModel },
+                // Set the data context for the main grid in the window.
+                MainGrid = { DataContext = viewModel },
 
-                    // Set the owner of the window to the Dynamo window.
-                    Owner = p.DynamoWindow
-                };
-
-                window.Left = window.Owner.Left + 400;
-                window.Top = window.Owner.Top + 200;
-
-                // Show a modeless window.
-                window.Show();
+                // Set the owner of the window to the Dynamo window.
+                Owner = p.DynamoWindow
             };
+
+            sampleMenuItem = new MenuItem { Header = "Show View Extension Sample Window", IsCheckable = true };
+            sampleMenuItem.Checked += (sender, args) => p.AddToExtensionsSideBar(this, window);
+            sampleMenuItem.Unchecked += (sender, args) => p.CloseExtensioninInSideBar(this);
```

Comment:
We have a few examples in Dynamo code base to unsubstribe in `dispose()`, would you also do that in sample?

Reply:
@QilongTang While we could, I think it also makes the sample harder to follow. For our "pre-installed" extensions we are doing that to make sure we dispose everything in our unit tests, where we create and dispose DynamoViews hundreds of times. This is not something that affects normal usage of Dynamo, where extensions get loaded only once on startup and disposed only once on shutdown.